src/BizHawk.Emulation.Cores/Consoles/Nintendo/Gameboy/Gambatte.cs

```diff
@@ -101,7 +101,27 @@ public Gameboy(CoreComm comm, GameInfo game, byte[] file, Gameboy.GambatteSettin
 
 				if (_syncSettings.EnableBIOS)
 				{
-					bios = comm.CoreFileProvider.GetFirmware(biosSystemId, biosId, true, "BIOS Not Found, Cannot Load.  Change SyncSettings to run without BIOS.");
+					try
+					{
+						bios = comm.CoreFileProvider.GetFirmware(biosSystemId, biosId, true, "BIOS Not Found, Cannot Load.  Change SyncSettings to run without BIOS.");
+					}
+					catch (MissingFirmwareException)
+					{
+						if (_syncSettings.ConsoleMode == GambatteSyncSettings.ConsoleModeType.GBA)
+						{
+							//implicitly throws MissingFirmwareException if GBC BIOS is also missing
+							bios = comm.CoreFileProvider.GetFirmware(biosSystemId, "World", true, "BIOS Not Found, Cannot Load.  Change SyncSettings to run without BIOS.");
+							byte[] agbOverride = { 0xFF, 0x00, 0xCD, 0x03, 0x35, 0xAA, 0x31, 0x90, 0x94, 0x00, 0x00, 0x00, 0x00 };
+							for (int i = 0xF3; i < 0x100; i++)
+							{
+								bios[i] = (byte)((agbOverride[i - 0xF3] + bios[i]) & 0xFF);

```

Comment:
Are you able to use XOR instead of addition? (And aren't the last 4 0x0 bytes of the patch redundant?)

Reply:
we should do whatever gambatte does

Reply:
Does gambatte (upstream) do anything?

Reply:
in upstream it edits the BIOS in the core, which we removed in our fork to support DMG BIOS and the fake BIOSes

Reply:
OK, I was not aware if whether the code I removed was an actual upstream thing.

In that case, we are doing exactly what upstream does.  I don't know why it does the extra 4 0x00 bytes, but they have no negative effect.